src/ray/raylet/worker_pool.cc

```diff
@@ -1470,7 +1468,23 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
                    << backlog_size << " and available CPUs " << num_available_cpus
                    << " num idle workers " << state.idle.size()
                    << " num registered workers " << state.registered_workers.size();
-    PrestartDefaultCpuWorkers(task_spec.GetLanguage(), num_needed);
+    PrestartWorkersWithRuntimeEnv(task_spec, num_needed);
+  }
+}
+
+void WorkerPool::PrestartWorkersWithRuntimeEnv(const TaskSpecification &task_spec,
+                                               int64_t num_needed) {
+  RAY_LOG(DEBUG) << "PrestartWorkers " << num_needed;
+  for (int ii = 0; ii < num_needed; ++ii) {
+    PopWorkerStatus status;
+    StartWorkerProcess(task_spec.GetLanguage(),
```

Comment:
I think we need to call `GetOrCreateRuntimeEnv(` first to update the reference counting.

Reply:
ok, I just found our invalid ref count decrement doesn't throw exception
https://github.com/ray-project/ray/blob/89990f6bc44e33d05b22fcc1a76baf0c9f09573c/python/ray/_private/runtime_env/agent/runtime_env_agent.py#L122-L128

Reply:
updated